listener/apischema/resolver.go

```diff
@@ -22,13 +22,13 @@ type Resolver interface {
 	Resolve(dc discovery.DiscoveryInterface, rm meta.RESTMapper) ([]byte, error)
 }
 
-type resolverImpl struct {
+type ResolverProvider struct {
```

Comment:
In Gateway we use Provider suffix for the struct which implements the interface.